harness/determined/cli/job.py

```diff
@@ -77,9 +77,10 @@ def computed_job_name(job: bindings.v1Job) -> str:
             j.type.value,
             computed_job_name(j) if isinstance(j, bindings.v1Job) else render.OMITTED_VALUE,
             j.priority if is_priority else j.weight,
-            pytz.utc.localize(
-                datetime.strptime(j.submissionTime.split(".")[0], "%Y-%m-%dT%H:%M:%S")
-            )
+            # submissionTime is of the format 2022-12-31T23:59:59.999999Z for DeterminedAI jobs and
+            # 2022-12-31T23:59:59Z for External jobs. We only need first 19 characters to
```

Comment:
I'm confused -- can we not parse this time when we receive it master side and standardize it?

Reply:
@stoksc both of these formats are valid variations of RFC3339 time, see [this section of the rfc](https://www.rfc-editor.org/rfc/rfc3339#section-5.6).

@jagadeesh545 this is also already solved in our code, see `det.common.util.parse_protobuf_timestamp()`.  Please use that instead, instead of either a 19-character hack or the `.split(".")` hack that was here previously.

Reply:
true. but it is weird a single api will return multiple variations to me.

Reply:
Updated as suggested.